enroll/endpoint.go

```diff
@@ -45,3 +82,44 @@ func MakeGetEnrollEndpoint(s Service) endpoint.Endpoint {
 		}
 	}
 }
+
+func MakeOTAEnrollEndpoint(s Service) endpoint.Endpoint {
+	return func(ctx context.Context, request interface{}) (interface{}, error) {
+		payload, err := s.OTAEnroll(ctx)
+		return mdmOTAEnrollResponse{payload, err}, nil
+	}
+}
+
+func MakeOTAPhase2Phase3Endpoint(s Service, scepDepot *boltdepot.Depot) endpoint.Endpoint {
+	return func(ctx context.Context, request interface{}) (interface{}, error) {
+		req := request.(mdmOTAPhase2Phase3Request)
+
+		// TODO: currently only verifying the signing certificate but ought to
+		// verify the whole provided chain. Note this will be difficult to do
+		// given the inconsist certificate chain returned by macOS in OTA mode,
+		// macOS in DEP mode, and iOS in either mode. See:
+		// https://openradar.appspot.com/radar?id=4957320861712384
+		if err := crypto.VerifyFromAppleDeviceCA(req.p7.GetOnlySigner()); err == nil {
+			// signing certificate is signed by the Apple Device CA. this means
+			// we don't yet have a SCEP identity and thus are in Phase 2 of the
+			// OTA enrollment
+			profile, err := s.OTAPhase2(ctx)
+			return mdmEnrollResponse{profile, err}, nil
+		} else if caChain, _, err := scepDepot.CA(nil); err == nil && len(caChain) > 0 && req.p7.GetOnlySigner().CheckSignatureFrom(caChain[0]) == nil {
```

Comment:
I'm not a fan of eating the error here if it's not nil. Is it possible for us to do something else? 

I'd also like to see you get rid of `if -> else if` since you `return` in each branch.